compat/maven-model-builder/src/main/java/org/apache/maven/model/building/ModelCache.java

```diff
@@ -18,23 +18,37 @@
  */
 package org.apache.maven.model.building;
 
-import java.util.function.Supplier;
-
-import org.apache.maven.building.Source;
-
 /**
  * Caches auxiliary data used during model building like already processed raw/effective models. The data in the cache
  * is meant for exclusive consumption by the model builder and is opaque to the cache implementation. The cache key is
  * formed by a combination of group id, artifact id, version and tag. The first three components generally refer to the
- * identity of a model. The tag allows for further classification of the associated data on the sole discretion of the
+ * identify of a model. The tag allows for further classification of the associated data on the sole discretion of the
  * model builder.
  *
  * @deprecated use {@link org.apache.maven.api.services.ModelBuilder} instead
  */
 @Deprecated(since = "4.0.0")
 public interface ModelCache {
 
-    <T> T computeIfAbsent(String groupId, String artifactId, String version, String tag, Supplier<T> data);
+    /**
+     * Puts the specified data into the cache.
+     *
+     * @param groupId The group id of the cache record, must not be {@code null}.
```

Comment:
Oracle javadoc guidelines are no initial capitalization on Javadoc comments and period only after or before a complete sentence